### PR TITLE
Added files with homework 091122

### PR DESCRIPTION
**Pull request by kate-lvlup**
Hello Dima,
Could you check my homework?
Thanks in advance!

P.S. I have the following questions related to my homework:

1. Почему double нельзя использовать с библиотекой Math (у меня падали эрроры когда я использовала double) ???
2. В таске4 я переделала так:
 int valuePow = 0;
 valuePow += Math.pow(valueSum, 2);

а изначально делала так, но падал эррор:
 int valuePow = 0;
 valuePow = valuePow + (Math.pow(valueSum, 2));

В чем отличие???

### CODE REVIEW

File: HomeworkLessonFour091122/LessonFourTaskFour.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+            valueArray[i] = (int)(1 + Math.random() * 9);
+            System.out.println("Array element [" + i + "]: " + valueArray[i]);
+        }
+        int valueSum = 0;
+        for (int i = 0; i < valueArray.length; i++) {
+            valueSum += valueArray[i];
+        }
+        int valuePow = 0;
+        valuePow += Math.pow(valueSum, 2);
+        System.out.println("Sum of array elements to the second power: " + valuePow);
```
Comment (YeaaaaH) [inline HomeworkLessonFour091122/LessonFourTaskFour.java:17]: i would use such approach:
int valueSum = 0;
for (int i = 0; i < valueArray.length; i++) {
    valueSum += Math.pow(valueArray[i], 2);
    System.out.println(valueSum);
}
System.out.println("Sum of array elements to the second power: " + valueSum);

File: HomeworkLessonFour091122/LessonFourTaskFive.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+class LessonFourTaskFive {
+    public static void main(String[] args) {
+        int[] valueArray;
+        valueArray = new int[10];
+        for (int i = 0; i < valueArray.length; i++) {
+            valueArray[i] = (int) (1 + Math.random() * 9);
+            System.out.println("Array element [" + i + "]: " + valueArray[i]);
+        }
+        int valueSum = 0;
```
Comment (YeaaaaH) [inline HomeworkLessonFour091122/LessonFourTaskFive.java:11]: this variable is redundant

File: HomeworkLessonFour091122/LessonFourTaskNine.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            valueArray[i] = (int)(Math.random() * 200 - 100);
+            System.out.println("Array element [" + i + "]: " + valueArray[i]);
+        }
+        int positiveSum = 0;
+        int negativeSum = 0;
+        for (int i = 0; i < valueArray.length; i++) {
+            if (valueArray[i] >= 0) {
+                positiveSum++;
+            } else {
+                negativeSum++;
```
Comment (YeaaaaH) [inline HomeworkLessonFour091122/LessonFourTaskNine.java:19]: this line increase amount of negative values, not calculating sum of them

File: HomeworkLessonFour091122/LessonFourTaskOne.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+        }
+        int valueSum = 0;
+        for (int i = 0; i < valueArray.length; i++) {
+            valueSum = valueSum + valueArray[i];
+        }
+        System.out.println("Sum of array values: " + valueSum);
+        int valueSumMath = 0;
+        for (int i = 0; i < valueArray.length; i++) {
+            valueSumMath = Math.addExact(valueSumMath, valueArray[i]);
+        }
```
Comment (YeaaaaH) [inline HomeworkLessonFour091122/LessonFourTaskOne.java:19]: this loop is redundant, you can calculate sum in previous loop, using Math.abs() function

File: HomeworkLessonFour091122/LessonFourTaskSix.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+// napisat' class kotoruy vuvodit v console massiv chisel v obratnom poryadke, polychennuy iz chisla vvedenogo v console
+
+import java.util.Scanner;
+
+class LessonFourTaskSix {
+    public static void main(String[] args) {
+        Scanner myScan = new Scanner(System.in);
+        System.out.println("Enter the number:");
+        int num = myScan.nextInt();
```
Comment (YeaaaaH) [inline HomeworkLessonFour091122/LessonFourTaskSix.java:9]: add if statement if entered value is less than 1

File: HomeworkLessonFour091122/LessonFourTaskThree.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        System.out.println("Enter the number:");
+        int num = myScan.nextInt();
+        int valueSum = 0;
+        if (num <= 0) {
+            System.out.println("You entered an invalid number");
+        }
+        if (num == 1) {
+            System.out.println("Sum: " + num);
+        }
+        if (num > 1) {
```
Comment (YeaaaaH) [inline HomeworkLessonFour091122/LessonFourTaskThree.java:17]: you can use if...else if...else here

File: HomeworkLessonFour091122/LessonFourTaskFour.java (hunk 1 above)
Comment (kate-lvlup) [reply HomeworkLessonFour091122/LessonFourTaskFour.java:17]: I used the method you described above but got the error

Comment (kate-lvlup) [reply HomeworkLessonFour091122/LessonFourTaskFour.java:17]: @YeaaaaH Could you share with me the correct code for this task (#4)?

Comment (kate-lvlup) [reply HomeworkLessonFour091122/LessonFourTaskFour.java:17]: Fixed

File: HomeworkLessonFour091122/LessonFourTaskNine.java (hunk 3 above)
Comment (kate-lvlup) [reply HomeworkLessonFour091122/LessonFourTaskNine.java:19]: Fixed

File: HomeworkLessonFour091122/LessonFourTaskFive.java (hunk 2 above)
Comment (kate-lvlup) [reply HomeworkLessonFour091122/LessonFourTaskFive.java:11]: fixed

File: HomeworkLessonFour091122/LessonFourTaskOne.java (hunk 4 above)
Comment (kate-lvlup) [reply HomeworkLessonFour091122/LessonFourTaskOne.java:19]: fixed

File: HomeworkLessonFour091122/LessonFourTaskSix.java (hunk 5 above)
Comment (kate-lvlup) [reply HomeworkLessonFour091122/LessonFourTaskSix.java:9]: fixed

**Review by YeaaaaH — APPROVED** [review]

**Comment by kate-lvlup** [comment]
All commits were merged into master branch.

**Pull request merged**